arm/proofs/instruction.ml

```diff
@@ -708,6 +708,26 @@ let arm_ADD = define
         let d:N word = word_add m n in
         (Rd := d) s`;;
 
+let arm_ADD_VEC = define
+ `arm_ADD_VEC Rd Rn Rm esize datasize =
+    \s. let n = read Rn s in
+        let m = read Rm s in
+        if datasize = 128 then
+          let d:(128)word =
+            if esize = 64 then simd2 word_add n m
+            else if esize = 32 then simd4 word_add n m
+            else if esize = 16 then simd8 word_add n m
+            else simd16 word_add n m in
+          (Rd := d) s
+        else
+          let n:(64)word = word_subword n (0,64) in
+          let m:(64)word = word_subword m (0,64) in
+          let d:(64)word =
+            if esize = 32 then simd2 word_add n m
+            else if esize = 16 then simd4 word_add n m
+            else simd8 word_add n m in
+          (Rd := word_zx d:(128)word) s`;;
```

Comment:
I used `word_zx` instead of `word_subword` even if the latter can be also used to extend the bitwidth, because I thought readers might not be aware of the fact that `word_subword` can extend the bitwidth.
There is one inconvenience about `word_zx` which is that it is not supported by tactics like `WORD_BLAST`, but applying `REWRITE_TAC[word_zx]` before using `WORD_BLAST` seems to work.

Reply:
Thanks for the observation! I believe I've now fixed the bug in WORD_BLAST (and related weaknesses around word_zx, word_sx) as a small addendum to this recent HOL Light update:

https://github.com/jrh13/hol-light/commit/6f6ac17e8f4bff6c7f6897de0570a004744e4963